### PR TITLE
feat: enable reduce_margin to return bool

### PR DESCRIPTION
**Pull request by yutakasi634**
The global interaction which I'm implementing now (https://github.com/Mjolnir-MD/Mjolnir/pull/236) needs the information that the neighbor list is updated or not in the current simulation step.
The interaction, which is based on GlobalInteractionBase, calls reduce_margin of SpatialPartiion, so, in this PR, I enable this reduce_margin method can notify that information via returned bool value.

### CODE REVIEW

**Comment by ToruNiina** [comment]
Lookes good! I think the same applies to the `scale_margin`, because it does almost the same thing. Do you have time to do that? Or I can send a PR to your (forked) repo.

**Comment by yutakasi634** [comment]
OK, I'll do that, so wait for a while.

**Comment by yutakasi634** [comment]
Thank you for your good suggestion. I've done!

File: mjolnir/core/SpatialPartitionBase.hpp [hunk 1]
```diff
@@ -112,11 +112,12 @@ class SpatialPartition
...
+
+    // reduce_margin return true if neighbour list is updated
+    bool reduce_margin(const real_type dmargin, const system_type& sys,
                        const potential_type& pot)
     {
-        partition_->reduce_margin(neighbors_, dmargin, sys, pot);
-        return ;
+        return partition_->reduce_margin(neighbors_, dmargin, sys, pot);
     }
     void scale_margin(const real_type scale, const system_type& sys,
```
Comment (ToruNiina) [inline mjolnir/core/SpatialPartitionBase.hpp:122]: It seems that you forgot to apply the change here.

**Review by ToruNiina — CHANGES_REQUESTED** [review]
It is almost okay, but I found a function that is overlooked.

File: mjolnir/core/SpatialPartitionBase.hpp (hunk 1 above)
Comment (ToruNiina) [reply mjolnir/core/SpatialPartitionBase.hpp:122]: The code snippet shown here is too confusing... I wanted to show the last line of this code fraction, the `scale_margin` function.

Comment (yutakasi634) [reply mjolnir/core/SpatialPartitionBase.hpp]: Thank you, I could understand and fixed that!

**Review by ToruNiina — APPROVED** [review]

**Pull request merged**